crates/weaver_live_check/src/attribute_live_check.rs

```diff
@@ -0,0 +1,727 @@
+            });
+        } else {
+            // Provide an info advice if the attribute is a template
+            if let Some(attribute) = &semconv_attribute {
+                if let AttributeType::Template(_) = attribute.r#type {
+                    attribute_result.add_advice(Advice {
+                        key: "template_attribute".to_owned(),
+                        value: Value::String(attribute.name.clone()),
+                        message: "Is a template".to_owned(),
+                        advisory: Advisory::Information,
+                    });
+                }
+            }
+        }
+
+        // run advisors on the attribute
+        for advisor in self.advisors.iter_mut() {
+            if let Ok(advices) = advisor.advise(sample_attribute, semconv_attribute.as_ref()) {
+                for advice in advices {
+                    attribute_result.add_advice(advice);
+                }
+            }
+        }
+
+        attribute_result
+    }
+
+    /// Run advisors on every attribute in the list
+    #[must_use]
+    pub fn check_attributes(&mut self, sample_attributes: Vec<SampleAttribute>) -> LiveCheckReport {
+        let mut live_check_report = LiveCheckReport {
+            attributes: Vec::new(),
+            statistics: LiveCheckStatistics::new(),
+        };
+
+        for sample_attribute in sample_attributes.iter() {
+            let attribute_result = self.create_live_check_attribute(sample_attribute);
+
+            // Update statistics
+            live_check_report.statistics.update(&attribute_result);
+            live_check_report.attributes.push(attribute_result);
+        }
+        live_check_report
+    }
+}
+
+/// Represents a live check attribute parsed from any source
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct LiveCheckAttribute {
+    /// The sample attribute
+    pub sample_attribute: SampleAttribute,
+    /// Advice on the attribute
+    pub all_advice: Vec<Advice>,
+    /// The highest advisory level
+    pub highest_advisory: Option<Advisory>,
+}
+
+impl LiveCheckAttribute {
+    /// Create a new LiveCheckAttribute
+    #[must_use]
+    pub fn new(sample_attribute: SampleAttribute) -> Self {
+        LiveCheckAttribute {
+            sample_attribute,
+            all_advice: Vec::new(),
+            highest_advisory: None,
+        }
+    }
+
+    /// Add an advice to the attribute and update the highest advisory level
+    pub fn add_advice(&mut self, advice: Advice) {
+        let advisory = advice.advisory.clone();
+        if let Some(previous_highest) = &self.highest_advisory {
+            if previous_highest < &advisory {
+                self.highest_advisory = Some(advisory);
+            }
+        } else {
+            self.highest_advisory = Some(advisory);
+        }
+        self.all_advice.push(advice);
+    }
+}
+
+/// A live check report for a set of attributes
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct LiveCheckReport {
```

Comment:
Shouldn’t we have the following structures in a separate file that isn’t solely associated with attributes? I’m thinking this `LiveCheckReport` will also be used to report advice on metrics, spans, events, etc.